src/components/media-loader.js

```diff
@@ -65,18 +72,19 @@ AFRAME.registerComponent("media-loader", {
       const { raw, images, contentType } = await resolveMedia(src, false, index);
 
       const isPDF = contentType.startsWith("application/pdf");
-      if (
-        contentType.startsWith("image/") ||
-        contentType.startsWith("video/") ||
-        contentType.startsWith("audio/") ||
-        isPDF
-      ) {
+      if (contentType.startsWith("video/") || contentType.startsWith("audio/")) {
+        this.el.removeAttribute("gltf-model-plus");
```

Comment:
Do we want to write `if (anythingChanged) {removeAllTheThings();}`? Perhaps we don't, because if the `src` of a `media-image` or `gltf-model-plus` changes, we don't _need_ to `remove` those components and create new ones.

Reply:
Yeah exactly. could maybe DRY it up into a function taht you give 1 component name and it remvoes all the others, but that doesnt really seem better.